### PR TITLE
Upgrade/app detail

### PR DESCRIPTION
**Pull request by leoendless**

### CODE REVIEW

File: src/components/Base/Form/form.jsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  constructor(props) {
+    super(props);
+    this._formData = props.data || {};
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if ('data' in nextProps) {
+      this._formData = nextProps.data || {};
```
Comment (iwisunny) [inline src/components/Base/Form/form.jsx]: since `this._formData` already has a default value in constructor, no need to assign `{}` here.

Comment (iwisunny) [reply src/components/Base/Form/form.jsx]: `data` in `nextProps ` may not be an object, should validate it firstly

Comment (iwisunny) [reply src/components/Base/Form/form.jsx]: `data` should be defined in propTypes

File: src/components/Base/Form/item.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  }
+
+  static defaultProps = {
+    className: '',
+  }
+
+  constructor(props) {
+    super(props);
+    const { formData, children } = this.props;
+    const { name } = children.props;
```
Comment (iwisunny) [inline src/components/Base/Form/item.jsx]: maybe the logic is
```javascript
const { name, value } = children.props;
if(name) {
  formData[name] = value || '';
}
```

File: src/components/Base/Form/form.jsx (hunk 1 above)
Comment (leoendless) [reply src/components/Base/Form/form.jsx]: ok

File: src/components/Base/Form/item.jsx (hunk 2 above)
Comment (leoendless) [reply src/components/Base/Form/item.jsx]: this part should be removed

**Pull request merged**